### PR TITLE
Add check for status change to prevent spam alerts, and associated test

### PR DESCRIPTION
**Pull request by JishnuM**
Fixes https://github.com/gyng/rcanary/issues/1.

Tested with unit test.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -161,6 +167,23 @@ fn check_host(target: &CanaryTarget) -> CanaryCheck {
...
+
+/*
+    Checks if alert would not be spam.
+    Alert would not be spam iff state changes from "okay" to "fire" or "fire" to "okay".
+    Updates HashMap last_statuses with last state seen for the target.
+*/
+fn check_not_spam(last_statuses: &mut HashMap<CanaryTarget, String>, result: &CanaryCheck) -> bool {
+    let not_spam = match last_statuses.get(&result.target) {
+        Some(status) => {
+            (status == "okay" && result.info == "fire") || (status == "fire" && result.info == "okay")
```
Comment (gyng) [inline src/main.rs]: `unknown` status is also not spam. I will change it to use a status enum in the future, but a check for an `unknown` string is needed here for now.

An easy way to test this is to run a local server with `python -m SimpleHTTPServer 8000`, and changing the config file to `host = "http://0.0.0.0:8000"`.

File: src/main.rs [hunk 2]
```diff
@@ -345,4 +369,75 @@ mod tests {
 
         assert_eq!(ok_expected, ok_actual);
     }
+
+    #[test]
+    fn it_checks_for_spam_alerts() {
```
Comment (gyng) [inline src/main.rs]: This large test should be split up into one test per case.

File: src/main.rs [hunk 3]
```diff
@@ -161,6 +167,23 @@ fn check_host(target: &CanaryTarget) -> CanaryCheck {
     }
 }
 
+
+/*
+    Checks if alert would not be spam.
+    Alert would not be spam iff state changes from "okay" to "fire" or "fire" to "okay".
+    Updates HashMap last_statuses with last state seen for the target.
+*/
+fn check_not_spam(last_statuses: &mut HashMap<CanaryTarget, String>, result: &CanaryCheck) -> bool {
```
Comment (JishnuM) [inline src/main.rs]: Will rewrite to `check_spam` removing the negation.

To confirm the expected behavior, if there is no previous history (`None` case), then the alert should be sent?

File: src/main.rs (hunk 2 above)
Comment (JishnuM) [inline src/main.rs]: Will do.

File: src/main.rs (hunk 1 above)
Comment (JishnuM) [inline src/main.rs]: I am not sure of the expected behavior now. The initial issue suggested that only changes from `Ok` to `Fire` and `Fire` to `Ok` are valid. Should the behavior instead be that *any* status change is not spam?

File: src/main.rs (hunk 3 above)
Comment (gyng) [inline src/main.rs]: The expected behaviour has been clarified [here](https://github.com/gyng/rcanary/pull/4#discussion_r83370871).

**Comment by gyng** [comment]
Please login to [govBuy](https://buy.gds-gov.tech) and mark your bid as done. We will contact you via email for payment matters.

**Pull request merged**